geoapps/drivers/components/data.py

```diff
@@ -0,0 +1,248 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from copy import deepcopy
+from typing import Dict, Tuple
+
+import numpy as np
+from scipy.interpolate import LinearNDInterpolator
+
+from geoapps.utils import filter_xy
+
+from .locations import InversionLocations
+
+
+class InversionData(InversionLocations):
+    """ Retrieve data from workspace and apply transformations. """
+
+    def __init__(self, workspace, params, mesh, topography, window):
+        super().__init__(workspace, params, mesh, window)
+        self.topography = topography
+        self.resolution = None
+        self.offset = None
+        self.radar = None
+        self.ignore_value = None
+        self.ignore_type = None
+        self.detrend_order = None
+        self.detrend_type = None
+        self.components = None
+        self.data = {}
+        self.uncertainties = {}
+        self.normalization = []
+        self.survey = None
+        self._initialize()
+
+    def _initialize(self):
+        """ Extract data from params class. """
+
+        self.ignore_value, self.ignore_type = self.parse_ignore_values()
+        self.components, self.data, self.uncertainties = self.get_data()
+
+        self.locs = super().get_locs(self.params.data_object)
+        self.mask = np.ones(len(self.locs), dtype=bool)
+
+        if self.window is not None:
+            self.mask = filter_xy(
+                self.locs[:, 0],
+                self.locs[:, 1],
+                window=self.window,
+                angle=self.angle,
+                mask=self.mask,
+            )
+
+        if self.params.resolution is not None:
```

Comment:
Same here

Reply:
Not sure which chunk you're referring to, but the resolution isn't shared by both topo and data, since I leave the topo sampling as is.  And the windowing is handled different in each too (topo copies the window and expands it's x/y limits

Reply:
Makes sense